pkg/controller/discovery/web/direct.go

```diff
@@ -123,6 +164,76 @@ func (h *DirectImageMigrationHandler) Prepare(ctx *gin.Context) int {
 	return http.StatusOK
 }
 
+//
+// Prepare to fulfil the request.
+// Fetch the referenced dism.
+// Perform SAR authorization.
+func (h *DirectImageStreamMigrationHandler) Prepare(ctx *gin.Context) int {
+	status := h.BaseHandler.Prepare(ctx)
+	if status != http.StatusOK {
+		return status
+	}
+	name := ctx.Param(DirectVolumeParam)
+	if name != "" {
+		h.directImageStream = model.DirectImageStreamMigration{
+			CR: model.CR{
+				Namespace: ctx.Param(NsParam),
+				Name:      ctx.Param(DirectImageStreamParam),
+			},
+		}
+		err := h.directImageStream.Get(h.container.Db)
+		if err != nil {
+			if err != sql.ErrNoRows {
+				Log.Trace(err)
+				return http.StatusInternalServerError
+			} else {
+				return http.StatusNotFound
+			}
+		}
+	}
+	status = h.allow(h.getSAR())
+	if status != http.StatusOK {
+		return status
+	}
+
+	return http.StatusOK
+}
+
+//
+// Prepare to fulfil the request.
+// Fetch the referenced dvmp.
+// Perform SAR authorization.
+func (h *DirectVolumeMigrationProgressHandler) Prepare(ctx *gin.Context) int {
+	status := h.BaseHandler.Prepare(ctx)
+	if status != http.StatusOK {
+		return status
+	}
+	name := ctx.Param(DirectVolumeParam)
```

Comment:
This needs to be _DirectVolumeProgressParam_.

Reply:
Thanks Jeff. Good catch.